src/models/string-or-translations.ts

```diff
@@ -2,18 +2,29 @@ import Joi from 'joi';
 import { schema as translationsSchema, Translations } from './translations';
 import { Locale } from './locale';
 
-const schema = Joi.alternatives().try(
-  Joi.string().example('untranslated string').meta({ className: 'untranslatedString' }),
-  translationsSchema,
-)
-  .tag('stringOrTranslations')
-  .meta({ className: 'stringOrTranslations' });
+const schema = (limit?: number): Joi.AnySchema => {
+  if (limit !== undefined) {
+    return Joi.alternatives().try(
+      Joi.string().example('untranslated string').meta({ className: 'untranslatedString' }).max(limit),
+      translationsSchema(limit),
+    )
+      .tag('stringOrTranslations')
+      .meta({ className: 'stringOrTranslations' });
+  }
+
+  return Joi.alternatives().try(
+    Joi.string().example('untranslated string').meta({ className: 'untranslatedString' }),
+    translationsSchema(),
+  )
+    .tag('stringOrTranslations')
+    .meta({ className: 'stringOrTranslations' });
+};
 
-const stringBeforeV7ElseStringOrTranslationSchema = (apiVersion: number): Joi.AnySchema => {
-  if (apiVersion <= 6) {
+const versionedStringOrStringOrTranslationSchema = (apiVersion: number): Joi.AnySchema => {
+  if (apiVersion < 9) {
```

Comment:
This looks like a bad idea. Aren't we now breaking backward compatibility for anyone who was still on the stringBeforeV7 part?

Reply:
1cd34ee